### PR TITLE
Add filtering functionality to the network graphs

### PR DESCRIPTION
**Pull request by js-konda**
filter_menu

Added the filter_menu option and the functionality.
Major changes to review:
> functions that dynamically add options to tom-select in template.html
> the filtering functionality in utils.js
> docstrings in network.py

### CODE REVIEW

**Comment by jhunpingco** [comment]
@js-konda can you remove the extraneous `console.log` and other debug artifacts from the CodeBase?

File: pyvis/templates/template.html [hunk 1]
```diff
@@ -1,374 +1,553 @@
...
+                      aria-label="Default select example"
+                      onchange="selectNode([value]);"
+                  id="select-node"
+                  placeholder="Select node..."
+                    >
+                      <option selected>Select a Node by ID</option>
+                      {% for node in nodes %}
+                      <option value="{{ node.id }}">{{node.id}}</option>
+                      {% endfor %} {#
+                      <option value="2">Two</option>
```
Comment (BerserkerGaruk) [inline pyvis/templates/template.html:179]: Remove testing comment

**Review by BerserkerGaruk — COMMENTED** [review]
Read through the changes, everything looks to be good, ran some of it locally. Only comment I have to add in addition to @jhunpingco is that there is some commented out testing code that I marked in the review, and that there should be some tests implemented to ensure the changes are being properly rendered.(which is being worked on) I'll accept and merge after the tests are commited to the branch.

**Comment by yanirmr** [comment]
Hello, sir @js-konda .
Your branch provided me with a great deal of assistance.
Absolutely, I plan to present a graph at a conference next week and have been searching for weeks for a filter tool that can be applied to Pyvis. Thank you very much!

When I tried to use your code, I encountered the following issues:
* In the months since your fork, Pyvis has made some progress. There are some basic features (such as shapes in graphs) that are supposed to be updated in your branch. 
* It appears that the droplist selection filter does not support UTF-8 characters. The result is the appearance of "????????-????" words.
* The filtering functionality works well, but the highlighting functionality only works when clicking on the graph. Are there any methods available for searching for nodes and simply zooming and highlighting them?

I can help with the Python-side of this code if it is relevant.

Best wishes!

**Pull request merged**